### ISSUE TITLE
4K UI too small

### ISSUE THREAD

**Issue by DookieAllan** [r3eckon/BNG-BeamLegalRacing#15]
Hopefully this hasn't been addressed already, but the UI in 4k is very small. I can't seem to find any documentation or options for increasing its size. 1080p looks perfect as far as the UI is concerned.

**Comment by r3eckon** [comment]
You should be able to change the UI resolution to 1080p in the game options > user interface > advanced > UI maximum resolution, unfortunately all my monitors are 1080p so I have no way to test higher resolutions. Let me know if this works, if not I'll see if I can add a scaling option somehow but this might take a while as all the UI apps are built for 1080p.

**Comment by DookieAllan** [comment]
Thanks for the quick follow-up. Unfortunately, my UI max resolution is already set to 1080p, and setting it lower only impacts the beam.ng UI and not the mod.

BTW I'm really enjoying the mod, thank you!

**Comment by r3eckon** [comment]
Oh you're talking about the imgui menus right? Like the "Player Stats" window on the top left of the screen? Because I checked and when I set UI maximum resolution to 720p the "BeamLR Main Menu" does get bigger but imgui menus stay small. Basically it works for UI apps but not imgui. There's a console function to change imgui scale however until I implement these to automatically load when you start the mod you'll have to do it manually. Console opens with the ~ key.

Scale the imgui: `extensions.ui_imguiUtils.changeUIScale(2)` 

This would make the ui twice as big. Replace '2' with whatever value fits best. Then the issue is that windows will be too small for the text size, you can resize them however this change will not stick because the imgui ini file is constantly overwritten by BeamNG, so the solution is to save/load a custom imgui ini file:

Save a custom imgui ini file: `extensions.ui_imgui.saveIniSettingsToDisk("settings/beamlr_imgui.ini")` 
Load a custom imgui ini file: `extensions.ui_imgui.loadIniSettingsFromDisk("settings/beamlr_imgui.ini")` 

This file will keep not only imgui window sizes but also their positions and docking states. This is the result with a scale of 2 on a 1080p screen, with the ui app resolution set to 720p. Too large for my taste but you get the idea.

![](https://i.imgur.com/mufvjKt.png)

To recap: once you're done resizing/placing windows so they work with the increased scale, use the command to save. Then when you start playing BeamLR, wait until the mod has finished loading and execute the scaling function and the ini file loading function. This should restore your layout. Also note that opening world editor will reset the scale value. Also, if you never use the lua console for anything else you don't have to copy/paste every time, just use the up arrow to access commands you previously used.

Thanks for reporting this issue, for next version I'll add an option that loads this stuff automatically when the mod is loading. And I'm glad you're enjoying the mod!
